### PR TITLE
Automatic discovery of sli examples in example runner

### LINKED ISSUE #1138: `run_examples.sh` can not find examples
This is just a smaller issue.

If you want to run `run_examples.sh` in an installation of NEST in `NEST_INSTALL_DIR/share/doc/nest/` it fails, because it can not find `../examples`.

You might be able to change this by adjusting `NEST_SOURCE/examples/CMakeLists.txt` in line 20, or changing `run_examples.sh` in line 50 - the latter would cause other problems again.

### PR DESCRIPTION
**Pull request by gtrensch**
This PR adds automatic selection of sli examples.

Fixes #1138.

### CODE REVIEW

File: examples/nest/ArtificialSynchrony.sli [hunk 1]
```diff
@@ -19,6 +19,9 @@
  *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
  *
  */
+
+ % autorun=true
```
Comment (jougs) [inline examples/nest/ArtificialSynchrony.sli]:
```suggestion
% autorun=true
```

File: examples/run_examples.sh [hunk 2]
```diff
@@ -67,31 +61,36 @@ for i in $EXAMPLES ; do
...
+    
     echo ">>> RUNNING: $workdir/$example"
+    echo "    LOGFILE: $logfile"
 
-    set +e
-    $runner $example
-
+    export NEST_DATA_PATH=$output_dir
+    /usr/bin/time -f "$time_format" --quiet sh -c "$runner $example >$logfile 2>&1"
+
```
Comment (jougs) [inline examples/run_examples.sh]:
```suggestion

```

File: examples/run_examples.sh [hunk 3]
```diff
@@ -19,41 +19,35 @@
...
-    *)
-        SKIP=$SKIP_LIST ;;
-esac
+FAILURES=0
 
-# Find all examples in the installation directory
-EXAMPLES=$(find ${SEARCH_DIR:-./examples} -type f -name \*.py -o -name \*.sli | sort -t. -k3)
+# Find all examples in the installation directory that have a line
+# containing "autorun=true"
+EXAMPLES=$(grep -rl --include=\*\.sli 'autorun=true' nest/)
```
Comment (steffengraber) [inline examples/run_examples.sh]: This only works in source code. In an installation `run_examples.sh` is in  `NEST_INSTALL_DIR/share/doc/nest/`. There the examples are in `examples/` not `nest/`. See #1138.

Comment (gtrensch) [reply examples/run_examples.sh]: @steffengraber I have addressed this, please re-review.

**Comment by gtrensch** [comment]
I have addressed all issues, please re-review.

**Comment by steffengraber** [comment]
Just tried and it works now and solve issue #1138.

**Pull request merged**